urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/User_Interface/ApplicationWindow.java

```diff
@@ -0,0 +1,981 @@
+package User_Interface;
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.MediaTracker;
+import java.awt.Point;
+import java.awt.event.*;
+import java.io.File;
+import java.io.IOException;
+
+import javax.print.attribute.standard.Media;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+import javax.swing.*;
+import javax.swing.border.TitledBorder;
+import javax.swing.plaf.basic.BasicProgressBarUI;
+import javax.swing.text.DefaultCaret;
+import Client_Server.AGClient;
+import Game_World.Avatar;
+import Game_World.ExternalDoor;
+import Game_World.GameWorld;
+import Game_World.InternalDoor;
+import Game_World.Altar;
+import Game_World_Objects.Chest;
+import Game_World_Objects.Food;
+import Game_World_Objects.Key;
+import Game_World_Objects.Relic;
+import Game_World_Objects.Torch;
+import Graphics_Renderer.GraphicsRenderer;
+import Junit_Tests.Class_Examples;
+import Object_Interfaces.Containable;
+import Object_Interfaces.Container;
+import Object_Interfaces.GameObject;
+
+/**
+ * This class is used to display the main graphical user interface for the game.
+ * This class interacts with GraphicsRenderer to display the current state of a
+ * game, as well as communicate with AGClient to keep the local state up to date
+ * with the server state.
+ *
+ * @author Alex Candler, 300257532, candlealex
+ *
+ */
+public class ApplicationWindow implements MouseListener, KeyListener,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 54 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.